src/orquestra/sdk/_base/_logs/_markers.py

```diff
@@ -0,0 +1,136 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+import json
+import re
+import sys
+import typing as t
+from contextlib import contextmanager
+from dataclasses import dataclass
+
+from orquestra.sdk.schema.ir import TaskInvocationId
+from orquestra.sdk.schema.workflow_run import WorkflowRunId
+
+from .. import _dates
+
+ORQ_MARKER_PREFIX = "ORQ-MARKER:"
+ORQ_MARKER_PATTERN = re.compile(re.escape(ORQ_MARKER_PREFIX) + r"(.+)")
+
+
+# `TaskStartMarker` and `TaskEndMarker` are record types that hold the information we
+# emit in the log markers. They're dataclasses and not pydantic models because we're
+# not just using plain JSON: we're using prefix markers.
+
+
+@dataclass(frozen=True)
+class TaskStartMarker:
+    event = "task_start"
+    wf_run_id: WorkflowRunId
+    task_inv_id: TaskInvocationId
+    timestamp: _dates.Instant
+
+    @property
+    def line(self) -> str:
+        event = {
+            "event": "task_start",
```

Comment:
Suggestion: should this be:
```suggestion
            "event":  self.event,
```